### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by ivlytvynets**

### CODE REVIEW

File: src/main/java/core/basesyntax/Figure.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        this.color = color;
+    }
+
+    public Color getColor() {
+        return color;
+    }
+
+    public abstract void setArea();
+
+    public abstract void getInfo();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java:18]: let's move this method to an interface, they are usually used to declare a behavior

File: src/main/java/core/basesyntax/Main.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureProducer figureProducer = new FigureProducer();
+        for (int i = 0; i < new Random().nextInt(1000); i++) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Main.java:8]: we should create an array with figures

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Circle.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
```
[the remark this answers is not in the document]
Comment (ivlytvynets) [reply src/main/java/core/basesyntax/Circle.java:3]: I didn't understand. What problem is this?

[the remark this answers is not in the document]
Comment (ivlytvynets) [reply src/main/java/core/basesyntax/Circle.java:3]: I have `protected double area` in class Figure

[the remark this answers is not in the document]
Comment (ivlytvynets) [reply src/main/java/core/basesyntax/Circle.java:3]: fixed

File: src/main/java/core/basesyntax/Figure.java (hunk 1 above)
Comment (ivlytvynets) [reply src/main/java/core/basesyntax/Figure.java:22]: fixed

File: src/main/java/core/basesyntax/Main.java (hunk 2 above)
Comment (ivlytvynets) [reply src/main/java/core/basesyntax/Main.java]: fixed

File: src/main/java/core/basesyntax/Circle.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        }
+        this.radius = radius;
+    }
+
+    public double getRadius() {
+        return radius;
+    }
+
+    @Override
+    public void setArea() {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Circle.java:19]:
```suggestion
    public void calculateArea() {
```
we use setter when we insert value into field, for example public void setArea(double area)

File: src/main/java/core/basesyntax/Figure.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        return color;
+    }
+
+    public double getArea() {
+        return area;
+    }
+
+    public abstract void setArea();
+
+    public abstract void getInfo();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java:22]: previous comment is still not fixed, getInfo method should be in the interface, same with calculateArea(), let;s remove it from here

File: src/main/java/core/basesyntax/FigureProducer.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureProducer {
+
+    private static final int COUNT_OF_FIGURE_IMPLEMENTATIONS = 5;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/FigureProducer.java:7]:
```suggestion
    private static final int COUNT_OF_FIGURE = 5;
```

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Figure.java (hunk 5 above)
Comment (ivlytvynets) [reply src/main/java/core/basesyntax/Figure.java:22]: fixed

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**